components/eamxx/src/share/io/scream_scorpio_interface.F90

```diff
@@ -1359,6 +1360,40 @@ subroutine get_pio_atm_file(filename,pio_file,purpose)
       endif
     endif
   end subroutine get_pio_atm_file
+!=====================================================================!
+  ! Retrieve the time value for a specific time_index
+  ! If the input arg time_index is <= 0 then it is assumed the user wants
+  ! the last time entry.
+  function read_time_at_index(filename,time_index) result(val)
+    use pio,          only: PIO_get_var
+    use pio_nf,       only: PIO_inq_varid
+    character(len=*), intent(in) :: filename
+    integer, intent(in)          :: time_index
+    real(c_double)               :: val
+    real(c_double)               :: val_buf(1)
+    
+    type(pio_atm_file_t), pointer :: pio_atm_file
+    logical                       :: found
+    integer                       :: ierr
+    type(var_desc_t)              :: varid ! netCDF variable ID
+    integer                       :: strt(1), cnt(1)
+
+    call lookup_pio_atm_file(trim(filename),pio_atm_file,found)
+    if (.not.found) call errorHandle("pio_inq_dimlen ERROR: File "//trim(filename)//" not found",-999)
+    ierr = PIO_inq_varid(pio_atm_file%pioFileDesc,"time",varid)
+    call errorHandle('read_time_at_index: Error finding variable ID for "time" in file '//trim(filename)//'.',ierr);
+
+    if (time_index .gt. 0) then
+      strt(1) = time_index
```

Comment:
Should we inquire the time dim to ensure the time_index is not too large (at least in debug build)?